aten/src/ATen/native/cpu/Activation.cpp

```diff
@@ -401,41 +401,80 @@ void hardswish_backward_kernel(TensorIterator& iter) {
+    Vectorized<float> negval_v = Vectorized<float>(negval);
     cpu_kernel_vec(
         iter,
-        [&](scalar_t a) -> scalar_t {
-          return a > scalar_t(0) ? a : a * negval;
+        [&](BFloat16 a) -> BFloat16 {
+          return float(a) > float(0) ? float(a) : float(a) * negval;
         },
-        [&](Vec a) -> Vec {
-          auto r = Vec::blendv(negval_v, one_vec, a > zero_vec);
-          return a * r;
+        [&](Vectorized<BFloat16> a) -> Vectorized<BFloat16> {
+          Vectorized<float> a0, a1;
+          std::tie(a0, a1) = convert_bfloat16_float(a);
+          auto res0 = a0 * (Vectorized<float>::blendv(negval_v, one_vec, a0 > zero_vec));
+          auto res1 = a1 * (Vectorized<float>::blendv(negval_v, one_vec, a1 > zero_vec));
+          return convert_float_bfloat16(res0, res1);
         });
-  });
+  } else {
+    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "leaky_relu_cpu", [&] {
+      using Vec = Vectorized<scalar_t>;
+      auto zero_vec = Vec((scalar_t)(0));
+      auto one_vec = Vec((scalar_t)(1));
+      scalar_t negval = negval_.to<scalar_t>();
+      Vec negval_v = Vec(negval);
+      cpu_kernel_vec(
+          iter,
+          [&](scalar_t a) -> scalar_t {
+            return a > scalar_t(0) ? a : a * negval;
+          },
+          [&](Vec a) -> Vec {
+            auto r = Vec::blendv(negval_v, one_vec, a > zero_vec);
+            return a * r;
+          });
+    });
+  }
 }
 
 static void leaky_relu_backward_kernel(TensorIteratorBase& iter, const Scalar& negval_) {
-  AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "leaky_relu_backward_cpu", [&] {
-    using Vec = Vectorized<scalar_t>;
-    auto zero_vec = Vec((scalar_t)(0));
-    auto one_vec = Vec((scalar_t)(1));
-    scalar_t negval = negval_.to<scalar_t>();
-    Vec negval_v = Vec(negval);
+  if (iter.common_dtype() == kBFloat16) {
+    auto zero_vec = Vectorized<float>((float)(0));
+    auto one_vec = Vectorized<float>((float)(1));
+    float negval = negval_.to<float>();
+    Vectorized<float> negval_v = Vectorized<float>(negval);
     cpu_kernel_vec(
-        iter,
-        [&](scalar_t a, scalar_t b) -> scalar_t {
-          return a > scalar_t(0) ? b : b * negval;
-        },
-        [&](Vec a, Vec b) -> Vec {
-          auto r = Vec::blendv(negval_v, one_vec, a > zero_vec);
-          return b * r;
-        });
-  });
+      iter,
+      [&](BFloat16 a, BFloat16 b) -> BFloat16 {
+        return float(a) > float(0) ? float(b) : float(b) * negval;
+      },
+      [&](Vectorized<BFloat16> a, Vectorized<BFloat16> b) -> Vectorized<BFloat16> {
+        Vectorized<float> a0, a1, b0, b1;
+        std::tie(a0, a1) = convert_bfloat16_float(a);
+        std::tie(b0, b1) = convert_bfloat16_float(b);
+        auto res0 = b0 * (Vectorized<float>::blendv(negval_v, one_vec, a0 > zero_vec));
+        auto res1 = b1 * (Vectorized<float>::blendv(negval_v, one_vec, a1 > zero_vec));
+        return convert_float_bfloat16(res0, res1);
+      });
+  } else {
```

Comment:
theoretically we don't need to separate `BFloat16` as a special path here, since it is neither consecutive arithmetic operations (e.g.  "a + b * c"), nor accumulation (e.g. "sum").

we should be able to reuse `Vectorized<scalar_t>` for `BFloat16` as well. `blendv` is carried out via `_mm256_blendv_epi8` so it won't required to be converted to float32 before the operation. And we should be able to save the fp32->bf16 conversion before writing back to output tensor. Also fp32->bf16 is more expensive than bf16->fp32 with the emulated implementation.

@yanbing-j , please compare the perf of the following two implementations:
1. current impl
2. directly reuse `Vectorized<scalar_t>` for `BFloat16`, the code should look similar to [threshhold_kernel](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cpu/Activation.cpp#L86), aka. relu.